src/Module/AdminEditServiceUiUpdateHandler.php

```diff
@@ -304,19 +325,36 @@ protected function _updateSessionRules($serviceId, $newRules)
      *
      * @param int|string|Stringable      $serviceId The ID of the service.
      * @param array|stdClass|Traversable $ruleData  The session rule data that was received.
+     * @param string|Stringable          $serviceTz The service timezone name.
      *
      * @return array|stdClass|ArrayAccess|ContainerInterface The processed session rule data.
      */
-    protected function _processSessionRuleData($serviceId, $ruleData)
+    protected function _processSessionRuleData($serviceId, $ruleData, $serviceTz)
     {
+        $allDay   = $this->_containerGet($ruleData, 'isAllDay');
+
+        // Parse the service timezone name into a timezone object
+        $timezoneName = $this->_normalizeString($serviceTz);
+        $timezone     = empty($timezoneName) ? null : new DateTimeZone($timezoneName);
+
+        // Get the start ISO 8601 string, parse it and normalize it to the beginning of the day if required
+        $startIso8601    = $this->_containerGet($ruleData, 'start');
+        $startDatetime   = Carbon::parse($startIso8601, $timezone);
```

Comment:
Is it possible to avoid use of non-injectable, undeclared dependency `Carbon`? Say, by using [`strtotime()`][]?

[`strtotime()`]: http://php.net/manual/en/function.strtotime.php

Reply:
Dayum. Ok. [Card](https://github.com/orgs/RebelCode/projects/2#card-12084281).